worlds/hk/__init__.py

```diff
@@ -218,11 +247,58 @@ def create_items(self):
             for loc, price in zip(locations, prices):
                 loc.cost = price
 
+    def _can_beat_thk(self, state, player):
+        return (
+            state.has('Opened_Black_Egg_Temple', player)
+            and (state.count('FIREBALL', player) + state.count('SCREAM', player) + state.count('QUAKE', player)) > 1
+            and (  # NAILCOMBAT
+                    state.has('LEFTSLASH', player)
+                    or state.has('RIGHTSLASH', player)
+                    or state.has('UPSLASH', player)
+            )
+            and (
+                    (state.has('LEFTDASH', player) or state.has('RIGHTDASH', player))
+                    or self.world.ProficientCombat[self.player]
+            )
+        )
+
     def set_rules(self):
         world = self.world
         player = self.player
         if world.logic[player] != 'nologic':
-            world.completion_condition[player] = lambda state: state.has('DREAMER', player, 3)
+            goal = world.Goal[player]
+            if goal == Goal.option_siblings:
+                world.completion_condition[player] = lambda state: (
+                    self._can_beat_thk(state, player)
+                    and state.has('WHITEFRAGMENT', player, 3)
+                )
```

Comment:
This could be set to use the `HKLogicMixin` if you move the `_can_beat_thk()` to the mixin class.

Reply:
This should now be done.  HKLogicMixin didn't exist when I first opened this, hopefully I used it correctly.